lib/phase/flatten.js

```diff
@@ -306,6 +306,46 @@ function flattenSharedIntoJobs(shared, jobs) {
     return newJobs;
 }
 
+/**
+ * Merge shared steps into job when mergeSharedSteps annotation is set to true
+ * @param  {Object} sharedConfig               Shared config
+ * @param  {Object} jobConfig                  Job config
+ * @param  {Object} [template]                 Template
+ * @return {Object}                            Merged job config
+ */
+function handleMergeSharedStepsAnnotation({ sharedConfig, jobConfig, template }) {
+    const sharedSteps = sharedConfig.steps;
+    const newJobConfig = jobConfig;
```

Comment:
for jobs that use a template, at this point, the steps need be expanded and treated as the steps of this job and subsequently merged with the shared steps if any. I think the current implementation does not handle the expansion of the steps from job template?